client/src/Components/ProductCard.jsx

```diff
@@ -1,31 +1,35 @@
+/* eslint-disable react/prop-types */
 /* eslint-disable no-unused-vars */
 import React from 'react';
 import '../assets/styles/cards.css';
-import { Link } from 'react-router-dom';
+import { Link, NavLink } from 'react-router-dom';
 
 function ProductCard({
   product: {
-    name, image, description, price, discount, details,
+    // eslint-disable-next-line react/prop-types
+    name, image, id, description, price, discount, details,
   },
 }) {
   return (
-    <div className="product-card">
-      <div className="product-tumb">
-        <img src={image} width="100%" alt="" />
-
-      </div>
-      <div className="product-details">
-        <span className="product-catagory">{details}</span>
-        <h4><a href="#link">{description}</a></h4>
-        <p>{details}</p>
-        <div className="product-bottom-details">
-          <div className="product-price">{price}</div>
-          <div className="product-links">
-            <Link href="#link"><i className="fa fa-shopping-cart" /></Link>
+    <>
+      <NavLink to={`product/${id}`}> link</NavLink>
```

Comment:
why do you use NavLink if you don't Need it 
you can use the Link component if you wouldn't add style to the NavLink 
Navlink have two class (active or not) we use it to style the navbar links using this class name